src/main/java/com/jme3/math/Quaternion.java

```diff
@@ -803,110 +803,115 @@ public float toAngleAxis(Vector3f axisStore) {
-		float scale1 = t;
-
-		// Check if the angle between the 2 quaternions was big enough to
-		// warrant such calculations
-		if ((1 - result) > 0.0f) {
-			// Get the angle between the 2 quaternions,
-			// and then store the sin() of that angle
-			float theta = FastMath.acos(result);
-			float invSinTheta = 1f / FastMath.sin(theta);
-
-			// Calculate the scale for q1 and q2, according to the angle and
-			// its sine value
-			scale0 = FastMath.sin((1 - t) * theta) * invSinTheta;
-			scale1 = FastMath.sin((t * theta)) * invSinTheta;
+		if (rw < 0) {
+			return this.pureSlerpLocal(q2.negate(), t);
+		} else {
+			return this.pureSlerpLocal(q2, t);
 		}
+	}
 
-		// Calculate the x, y, z and w values for the quaternion by using a
-		// special
-		// form of linear interpolation for quaternions.
-		this.x = (scale0 * q1.x) + (scale1 * q2.x);
-		this.y = (scale0 * q1.y) + (scale1 * q2.y);
-		this.z = (scale0 * q1.z) + (scale1 * q2.z);
-		this.w = (scale0 * q1.w) + (scale1 * q2.w);
 
-		// Return the interpolated quaternion
-		return this;
+	public Quaternion pureSlerp(Quaternion q1, Quaternion q2, float t) {
+		return set(q1).pureSlerpLocal(q2, t);
 	}
 
 	/**
-	 * Sets the values of this quaternion to the slerp from itself to q2 by
-	 * changeAmnt
+	 * <code>slerp</code> sets this quaternion's value as an interpolation
+	 * between two other normalized quaternions.
 	 *
-	 * @param q2 Final interpolation value
-	 * @param changeAmnt The amount diffrence
+	 * @param q1 the first quaternion.
+	 * @param q2 the second quaternion.
+	 * @param t the amount to interpolate between the two quaternions.
 	 */
-	public void slerp(Quaternion q2, float changeAmnt) {
-		if (this.x == q2.x && this.y == q2.y && this.z == q2.z && this.w == q2.w) {
-			return;
-		}
-
-		float result = (this.x * q2.x) + (this.y * q2.y) + (this.z * q2.z) + (this.w * q2.w);
-
-		if (result < 0.0f) {
-			// Negate the second quaternion and the result of the dot product
-			q2.x = -q2.x;
-			q2.y = -q2.y;
-			q2.z = -q2.z;
-			q2.w = -q2.w;
-			result = -result;
-		}
+	public Quaternion slerp(Quaternion q1, Quaternion q2, float t) {
+		float rw = q1.w * q2.w + q1.x * q2.x + q1.y * q2.y + q1.z * q2.z;
 
-		// Set the first and second scale for the interpolation
-		float scale0 = 1 - changeAmnt;
-		float scale1 = changeAmnt;
-
-		// Check if the angle between the 2 quaternions was big enough to
-		// warrant such calculations
-		if ((1 - result) > 0.0f) {
-			// Get the angle between the 2 quaternions,
-			// and then store the sin() of that angle
-			float theta = FastMath.acos(result);
-			float invSinTheta = 1f / FastMath.sin(theta);
-
-			// Calculate the scale for q1 and q2, according to the angle and
-			// its sine value
-			scale0 = FastMath.sin((1 - changeAmnt) * theta) * invSinTheta;
-			scale1 = FastMath.sin((changeAmnt * theta)) * invSinTheta;
+		if (rw < 0) {
+			return set(q1).pureSlerpLocal(q2.negate(), t);
+		} else {
+			return set(q1).pureSlerpLocal(q2, t);
```

Comment:
Why not use this instead to reduce code duplication?
```suggestion
			return pureSlerp(q1, q2.negate(), t);
		} else {
			return pureSlerp(q1, q2, t);
```